app/test/analyzer/handlers_test.dart

```diff
@@ -0,0 +1,178 @@
+        registerAnalysisBackend(new MockAnalysisBackend());
+        expectNotFoundResponse(
+            await issueGet('/packages/pkg_foo/1.2.3/unknown'));
+      });
+
+      scopedTest('/packages/pkg_foo/1.2.3/1', () async {
+        registerAnalysisBackend(new MockAnalysisBackend());
+        expectNotFoundResponse(await issueGet('/packages/pkg_foo/1.2.3/1'));
+      });
+    });
+
+    group('get analysis', () {
+      scopedTest('/packages/pkg_foo', () async {
+        registerAnalysisBackend(new MockAnalysisBackend());
+        expectJsonResponse(await issueGet('/packages/pkg_foo'), body: {
+          'packageName': 'pkg_foo',
+          'packageVersion': '1.2.3',
+          'analysis': 242345,
+          'timestamp': '2017-06-26T12:48:00.000',
+          'analysisVersion': '0.2.0',
+          'analysisStatus': 'success',
+          'analysisContent': null
+        });
+      });
+
+      scopedTest('/packages/pkg_foo?full=true', () async {
+        registerAnalysisBackend(new MockAnalysisBackend());
+        expectJsonResponse(await issueGet('/packages/pkg_foo?full=true'),
+            body: {
+              'packageName': 'pkg_foo',
+              'packageVersion': '1.2.3',
+              'analysis': 242345,
+              'timestamp': '2017-06-26T12:48:00.000',
+              'analysisVersion': '0.2.0',
+              'analysisStatus': 'success',
+              'analysisContent': {'content': 'from-pana'},
+            });
+      });
+
+      scopedTest('/packages/pkg_foo/1.2.3', () async {
+        registerAnalysisBackend(new MockAnalysisBackend());
+        expectJsonResponse(await issueGet('/packages/pkg_foo/1.2.3'), body: {
+          'packageName': 'pkg_foo',
+          'packageVersion': '1.2.3',
+          'analysis': 242345,
+          'timestamp': '2017-06-26T12:48:00.000',
+          'analysisVersion': '0.2.0',
+          'analysisStatus': 'success',
+          'analysisContent': null
+        });
+      });
+
+      scopedTest('/packages/pkg_foo/1.2.3/242345', () async {
+        registerAnalysisBackend(new MockAnalysisBackend());
+        expectJsonResponse(await issueGet('/packages/pkg_foo/1.2.3/242345'),
+            body: {
+              'packageName': 'pkg_foo',
+              'packageVersion': '1.2.3',
+              'analysis': 242345,
+              'timestamp': '2017-06-26T12:48:00.000',
+              'analysisVersion': '0.2.0',
+              'analysisStatus': 'success',
+              'analysisContent': null
+            });
+      });
+    });
+
+    group('trigger analysis', () {
+      scopedTest('/packages/pkg', () async {
+        final ReceivePort rp = new ReceivePort();
+        final Future<Task> taskFuture = rp.first;
+        registerTaskSendPort(rp.sendPort);
+        expectJsonResponse(await issuePost('/packages/pkg'),
+            body: {'status': 'OK'});
+        final Task task = await taskFuture;
+        expect(task.package, 'pkg');
+        expect(task.version, isNull);
```

Comment:
It closes it (ReceivePort is a non-broadcast stream), but better to be explicit about it.